ext/NormalizingFlowsEnzymeCoreExt.jl

```diff
@@ -0,0 +1,39 @@
+module NormalizingFlowsEnzymeCoreExt
+
+using EnzymeCore
+using NormalizingFlows
+using NormalizingFlows: ADTypes, DifferentiationInterface
+
+# deal with Enzyme readonly error: see https://discourse.julialang.org/t/enzyme-autodiff-readonly-error-and-working-with-batches-of-data/123012
+function NormalizingFlows._prepare_gradient(loss, adbackend::ADTypes.AutoEnzyme, θ, args...)
+    if isempty(args)
```

Comment:
Why distinguish this case? The `map` version would work equally well with no `args`.
To force specialization, see https://docs.julialang.org/en/v1.11/manual/performance-tips/#Be-aware-of-when-Julia-avoids-specializing